### ISSUE TITLE
Make 'state' externally read-only

### ISSUE THREAD

**Issue by jrmlhermitte** [bluesky/bluesky#959]
As @mrakitin pointed out, state should be property not just a string:

https://github.com/NSLS-II/bluesky/blob/master/bluesky/run_engine.py#L517

**Comment by mrakitin** [comment]
Thanks for documenting it!

**Comment by tacaswell** [comment]
I thought the state was controlled by a super state machine instance....

**Comment by tacaswell** [comment]
https://github.com/NSLS-II/bluesky/blob/4bedbe7f8659beece0c24716c2d885f673dd8031/bluesky/run_engine.py#L184-L186

**Comment by jrmlhermitte** [comment]
but should it be settalbe by user?
ie.
```py
_state = LoggingPropertyMachine(RunEngineStateMachine)

@property
def state(self):
    return self._state
```

**Comment by danielballan** [comment]
This and #960 might be a good enough occasion to rip out `super_state_machine` and replace it with something simpler.

**Comment by danielballan** [comment]
Closed by #1188
